src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -410,8 +407,29 @@ public SnapshotAccessRequestResponse approveRequest(UUID id) {
 
   private SnapshotAccessRequestResponse convertModelToApiResponse(
       SnapshotAccessRequestModel model) {
-    return model.toApiResponse(
-        snapshotBuilderSettingsDao.getBySnapshotId(model.sourceSnapshotId()));
+    return model.toApiResponse();
+  }
+
+  private SnapshotAccessRequestDetailsResponse generateModelDetails(
+      AuthenticatedUserRequest userRequest, SnapshotAccessRequestModel model) {
+    List<Integer> conceptIds = model.generateConceptIds();
+    SnapshotBuilderSettings settings =
+        snapshotBuilderSettingsDao.getBySnapshotId(model.sourceSnapshotId());
+    Map<Integer, String> concepts =
+        conceptIds.size() > 0
```

Comment:
Personally I prefer `.isEmpty()` over comparisons with zero
```suggestion
        !conceptIds.isEmpty()
```